src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -15,7 +15,7 @@ internal static class HandshakeHelpers
         /// <summary>
         /// Gets request headers needed process the handshake on the server.
         /// </summary>
-        public static readonly IEnumerable<string> NeededHeaders = new[]
+        public static readonly string[] NeededHeaders = new[]
```

Comment:
PS: We should never do this. It's just inefficient and prevents the JIT from optimizing.

Reply:
+1

I guess it was done to prevent the array from ever being modified.

Reply:
I'm gonna have to find all of the code that does this and fix it. Allocation profiles show interesting things. This also tends to happen when we resolve `IEnumerable<T>` from the DI container. Thought that one is harder to generalize, in hot code paths, it might be worth try-casting to `T[]` and falling back to `IEnumerable<T>` or `IList<T>` if used many times.